mindmeld/components/entity_resolver.py

```diff
@@ -562,38 +1151,575 @@ def _predict_exact_match(self, entity):
+                    result = sorted(zip(synonyms.keys(), similarity_scores), key=lambda x: x[1],
+                                    reverse=True)
+                results.append(result)
+
+        if is_single:
+            return results[0]
+
+        return results
+
+    def _fit(self, clean, entity_map):
+
+        if clean and os.path.exists(self.cache_path):
+            os.remove(self.cache_path)
+
+        # load mapping.json data and process it
+        augment_lower_case = self._runtime_configs["augment_lower_case"]
+        self._processed_entity_map = self._process_entity_map(
+            self.type,
+            entity_map,
+            augment_lower_case=augment_lower_case
+        )
+
+        # load embeddings from cache if exists, encode any other synonyms if required
+        synonyms, synonyms_embs = OrderedDict(), np.empty(0)
+        if os.path.exists(self.cache_path):
+            logger.info("Cached embs exists for entity %s. "
+                        "Loading existing data from: %s",
+                        self.type, self.cache_path)
+            cached_data = self._load_embeddings(self.cache_path)
+            synonyms, synonyms_embs = cached_data["synonyms"], cached_data["synonyms_embs"]
+        new_synonyms_to_encode = [syn for syn in self._processed_entity_map["synonyms"] if
+                                  syn not in synonyms]
+        if new_synonyms_to_encode:
+            new_synonyms_encodings = (
+                self.encode(
+                    new_synonyms_to_encode,
+                    batch_size=self.batch_size,
+                    concat_last_n_layers=self._runtime_configs["concat_last_n_layers"],
+                    normalize_token_embs=self._runtime_configs["normalize_token_embs"],
+                )
+            )
+            synonyms_embs = new_synonyms_encodings if not synonyms else np.concatenate(
+                [synonyms_embs, new_synonyms_encodings])
+            synonyms.update(
+                OrderedDict(zip(
+                    new_synonyms_to_encode,
+                    np.arange(len(synonyms), len(synonyms) + len(new_synonyms_to_encode)))
+                )
+            )
+
+        # encode artificial synonyms if required
+        aug_avg_syn_embs = self._runtime_configs["augment_average_synonyms_embeddings"]
+        dummy_new_synonyms_to_encode, dummy_new_synonyms_encodings = [], []
+        if aug_avg_syn_embs:
+            # obtain cnames to synonyms mapping
+            entity_mapping_synonyms = self._processed_entity_map["synonyms"]
+            cnames2synonyms = {}
+            for syn, cnames in entity_mapping_synonyms.items():
+                for cname in cnames:
+                    items = cnames2synonyms.get(cname, [])
+                    items.append(syn)
+                    cnames2synonyms[cname] = items
+            # assert dummy synonyms
+            for cname, syns in cnames2synonyms.items():
+                dummy_synonym = f"{cname} - SYNONYMS AVERAGE"
```

Comment:
this code block is very similar to the other code block in `1645`, lets create a helper function that both functions can call to create these dummy vars

Reply:
Sure, will do that. Thanks for pointing it out!

Reply:
Hi @vijay120 , I observed that although both share some common code snippet, but because the core logic of mean/max pooling and stacking ()using numpy vs scipy.sparse) is different, using a helper function will only make the code complex i think. Let me know what you think.

Reply:
Got it.